libc/sem.cc

```diff
@@ -27,7 +55,7 @@ OSV_LIBC_API
 int sem_init(sem_t* s, int pshared, unsigned val)
 {
     static_assert(sizeof(indirect_semaphore) <= sizeof(*s), "sem_t overflow");
-    new (s) indirect_semaphore(val);
+    from_libc(s) = std::make_unique<posix_semaphore>(val, 1, false);
```

Comment:
Why did you change the new(s) to a regular assignment here? I think this change is wrong (but maybe I'm wrong?). The regular assignment operator assumes that the left-hand side is already a valid object and needs to be destructed before being assigned over, but this is not what we want to do here - we want to create a new object.

Reply:
As it was, that line was causing a compiler error because it was not matching the constructor for the `indirect_semaphore` alias. I opted for the `make_unique` constructor here, but I can see why that might be an issue if it requires the the equal operator for assignment. I have changed this here to construct a new `posix_semaphore`, then use the new assignment for `s` with that new semaphore.